app/lib/utils/queryMapper.js

```diff
@@ -25,39 +25,77 @@ function getLocationHeading(query) {
   return undefined;
 }
 
-function getResultsHeading(query, loc) {
-  if (loc && query.type) {
-    const location = loc.toUpperCase();
-    if (isProfessionalChoice(query)) {
-      return `Sexual health professionals near '${location}'`;
-    }
-    if (utils.areEqual(query.type, constants.serviceTypes.kit)) {
-      if (query.origin === constants.serviceChoices['16to25']) {
-        return `Where you can pick up a free test kit near '${location}'`;
-      } else if (query.origin === constants.serviceChoices.over25) {
-        return `Where you can buy a test near '${location}'`;
-      }
-    }
+function getResultsInfoForProfessionalsByAge(ageChoice, location) {
+  const correctResultsParams = true;
+  const resultsHeading = `Sexual health professionals near '${location}'`;
+  const resultsExplanation = 'Here is a list of places where you can get tested by a sexual health professional.';
+  let resultsOnwardsJourneyPartial;
+  if (ageChoice === constants.serviceChoices['16to25']) {
+    resultsOnwardsJourneyPartial = 'includes/onwardsJourneyProfessional16to25.nunjucks';
   }
-  return undefined;
+  if (ageChoice === constants.serviceChoices.over25) {
+    resultsOnwardsJourneyPartial = 'includes/onwardsJourneyProfessionalOver25.nunjucks';
+  }
+  return {
+    correctResultsParams,
+    resultsExplanation,
+    resultsHeading,
+    resultsOnwardsJourneyPartial,
+  };
 }
 
-function getResultsExplanation(query) {
-  if (query.type) {
+function getResultsInfoForKitsByAge(ageChoice, location) {
+  let correctResultsParams;
+  let resultsExplanation;
+  let resultsHeading;
+  let resultsOnwardsJourneyPartial;
+  if (ageChoice === constants.serviceChoices['16to25']) {
```

Comment:
I'm not keen on using `let` when a `const` could be used - the logic might be clearer if the `let`s were removed and each `if` statement returned an object instead.

It would also be clearer what is returned when either of those cases are not matched, at the moment correctResultsParam is either `true` or `undefined`?